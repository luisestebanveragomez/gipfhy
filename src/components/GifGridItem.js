import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {

  return (
      <div className="card">
          <img src={ url } alt={ title } />
          <strong> { title } </strong>
      </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
