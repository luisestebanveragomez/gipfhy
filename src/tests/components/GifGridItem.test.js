import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('pruebas en <GifGridItem />', () => {

  const title = 'title';
  const url = 'hey';

  const wrapper = shallow(<GifGridItem title={ title } url={ url } />)

  test('debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('debe de tener un strong con el title', () => {
    const strong = wrapper.find('strong');
    expect( strong.text().trim() ).toBe( title );
  })

  test('debe de tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    expect( img.prop('src') ).toBe( url );
    expect( img.prop('alt') ).toBe( title );
  })
})