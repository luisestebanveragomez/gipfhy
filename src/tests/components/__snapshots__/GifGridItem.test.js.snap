// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="title"
    src="hey"
  />
  <strong>
     
    title
     
  </strong>
</div>
`;
